code/game/objects/items.dm

```diff
@@ -781,6 +783,8 @@
 /obj/item/proc/item_action_slot_check(slot, mob/user, datum/action/action)
 	if(slot & (ITEM_SLOT_BACKPACK|ITEM_SLOT_LEGCUFFED)) //these aren't true slots, so avoid granting actions there
 		return FALSE
+	if(!isnull(action_slots) && !(slot & action_slots))
+		return FALSE
```

Comment:
Looking through these, an overwhelming majority of items just have action_slots = slot_flags. 

It'd save a lot of developer headaches if we just defaulted action_slots to that?

```dm
/obj/item/proc/item_action_slot_check(slot, mob/user, datum/action/action)
	if(slot & (ITEM_SLOT_BACKPACK|ITEM_SLOT_LEGCUFFED)) 
		return FALSE
	if(isnull(action_slots)
		if(slot & slot_flags))
			return TRUE
	else
		if(slot & action_slots)
			return TRUE
	return FALSE
```

Organs can just be handled by an override of the proc

```dm
/obj/item/organ/item_action_slot_check(slot, mob/user, datum/action/action)
	return owner == user
```

Reply:
the issue with this is that it will prevent any items like haunted swords which can be placed in a slot but can be used wherever from functioning

Reply:
1. I'm pretty sure setting `action_slots = NONE` would work for that
2. Things can still override the proc to do custom  behavior like that